SYCL/ESIMD/api/functional/operators/operator_assignment.hpp

```diff
@@ -0,0 +1,96 @@
+//===-- operator_assignment.hpp - Functions for tests on simd assignment
+//      operators. --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd assignment operators.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+// The test proxy is used to verify the move assignment was called actually.
+#define __ESIMD_ENABLE_TEST_PROXY
+
+#include "../common.hpp"
+#include "../shared_element.hpp"
+
+namespace esimd_test::api::functional::ctors {
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename DataT, int NumElems, typename TestCaseT> class run_test {
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+    const std::vector<DataT> ref_data = generate_ref_data<DataT, NumElems>();
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+
+    shared_element<bool> was_moved(queue);
```

Comment:
Looks like the naming should be changed. It stores some `success` now, right? So for both copy and move assignment `true` is expected.
Once again, it's better to have some initial value provided explicitly

```suggestion
    shared_element<bool> is_correct_operator(queue, false);
```

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/762/commits/92df51c1386dfd2e52709190f04f2c7baaec113d#diff-edf690369570284c343904e24bdf038b8c4bb8e6b620e467934162c5cdd1e05eR60)